web/src/context/installer.jsx

```diff
@@ -52,18 +72,41 @@ const BUS_ADDRESS_FILE = "/run/agama/bus.address";
   */
 function InstallerClientProvider({ client, children }) {
   const [value, setValue] = useState(client);
+  const [retry, setRetry] = useState(0);
+  const [error, setError] = useState(false);
+
+  const resetClient = useCallback(async () => {
+    setValue(undefined);
+    console.warn(`Trying to connect to D-Bus (attempt ${retry + 1})`);
+    await new Promise(resolve => setTimeout(resolve, RETRY_INTERVAL));
+    setRetry(retry + 1);
+
+    if (retry > RETRIES) setError(true);
+  }, [setError, setRetry, retry]);
 
   useEffect(() => {
-    if (client !== undefined) {
-      const file = cockpit.file(BUS_ADDRESS_FILE);
-      file.read().then(address => {
-        setValue(createClient(address));
-      });
+    const connectClient = async () => {
+      const newClient = client || await createDefaultClient();
+
+      const connected = await newClient.isConnected();
+      if (connected) {
+        setValue(newClient);
+      } else {
+        resetClient();
+      }
+    };
+
+    if (value === undefined) {
+      connectClient();
     }
-  }, [client]);
+  }, [client, value, resetClient, retry]);
```

Comment:
Perhaps defining `#resetClient` inside the `useEffect` saves some re-renderings, because the current callback is redefine every time it is called (it changes its `retry` dependency). So, calling to `resetClient` forces a render.